tests/SwagLanguagePackTest.php

```diff
@@ -46,4 +50,54 @@ public function testThatPluginIsInstalled(): void
 
         static::assertNotNull($plugin, 'Plugin needs to be installed to run testsuite');
     }
+
+    public function testUpdateSnippetsAndLanguagesOnUpdate(): void
+    {
+        $connection = $this->getContainer()->get(Connection::class);
+
+        $lastLocale = SwagLanguagePack::SUPPORTED_LANGUAGES;
+        $lastLocale = \array_shift($lastLocale);
+
+        $languageCountBefore = $connection->fetchOne('SELECT COUNT(*) FROM `language`');
+        $swagLanguageCountBefore = $connection->fetchOne('SELECT COUNT(*) FROM `swag_language_pack_language`');
+
+        $languageId = $connection->fetchOne(
+            <<<'SQL'
+                SELECT `language`.`id` FROM `locale`
+                    LEFT JOIN `language` ON `language`.`locale_id` = `locale`.`id`
+                    WHERE `locale`.`code` = :code
+            SQL,
+            ['code' => $lastLocale]
+        );
+
+        $connection->executeStatement(
+            'DELETE FROM `swag_language_pack_language` WHERE `language_id` = :languageId',
+            ['languageId' => $languageId]
+        );
+
+        $connection->executeStatement(
+            'DELETE FROM `language` WHERE `id` = :languageId',
+            ['languageId' => $languageId]
+        );
+
+        static::assertNotEquals($languageCountBefore, (int) $connection->fetchOne('SELECT COUNT(*) FROM `language`'));
+        static::assertNotEquals($swagLanguageCountBefore, (int) $connection->fetchOne('SELECT COUNT(*) FROM `swag_language_pack_language`'));
+
+        $pluginLifeCycleService = $this->getContainer()->get(PluginLifecycleService::class);
+
+        $criteria = new Criteria();
+        $criteria->addFilter(new EqualsFilter('baseClass', SwagLanguagePack::class));
+
+        $pluginRepository = $this->getContainer()->get(\sprintf('%s.repository', PluginDefinition::ENTITY_NAME));
```

Comment:
The pluginRepository is created during setup you can use `$this->pluginRepository`